lib/new_relic/telemetry_sdk/buffer.rb

```diff
@@ -0,0 +1,50 @@
+# encoding: utf-8
+# frozen_string_literal: true
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-telemetry-sdk-ruby/blob/main/LICENSE for complete details.
+
+module NewRelic
+  module TelemetrySdk
+    class Buffer
+      attr_reader :items
+      attr_accessor :common_attributes
+
+      CAPACITY = 2000
+
+      def initialize common_attributes=nil
+        @items = []
+        @common_attributes = common_attributes
+        @lock = Mutex.new
+        @capacity = capacity
+      end
+
+      # Items recorded into the buffer must have a to_h method for transformation
+      def record item
+        @lock.synchronize do
+          if size < capacity
+            @items << item
+          end
```

Comment:
We will want to log if an item cannot be added to the buffer because it's at capacity. We should keep that in mind for https://github.com/newrelic/newrelic-telemetry-sdk-ruby/issues/11.

Reply:
Are we truly wanting to just skip adding vs. say, popping the oldest item and then adding the new item?

Reply:
Also, why not use a Queue like we did in Infinite Tracing?  This should allow adding and popping items from the `@items` without wrapping in a Mutex.

Reply:
> Also, why not use a Queue like we did in Infinite Tracing?

Especially now that we're no longer implementing buffer limits, we're not really interested in popping things off of `@items` - we are interested in the entire contents as an array, and we need to [format the entire contents](https://github.com/newrelic/newrelic-telemetry-sdk-ruby/blob/3cda72ffb5f1ee9ecdad6f717e66a7b4dfafac7b/lib/new_relic/telemetry_sdk/buffer.rb#L26) as well at some point. It seems simpler to do both of those things if `@items` is already an array.